archaius2-core/src/main/java/com/netflix/archaius/DefaultAppConfig.java

```diff
@@ -253,8 +224,8 @@ public DefaultAppConfig(Builder builder) {
                     .withStrInterpolator(getStrInterpolator())
                     .build();
                 
-            super.addConfig(loader.newLoader().withName(APPLICATION_LAYER).load(builder.configName));
-            super.addConfig(library = new CascadingCompositeConfig(LIBRARY_LAYER));
+            root.addConfig(loader.newLoader().withName(APPLICATION_LAYER).load(builder.configName));
```

Comment:
Agreed.  I'll make it so you can pass in your own Config for the application layer.